delegator/src/sign/sign.service.ts

```diff
@@ -21,14 +21,15 @@ export class SignService {
 
   async create(data: SignDto) {
     const transaction = await this.prisma.transaction.create({ data })
+    let validatedQuery
```

Comment:
I think the whole function should be inside of `try-catch` block. Each function could theoretically raise an error, we should be able to catch all. Also, it makes it easier to read.

```typescript
async create(data: SignDto) {
  try {
    const transaction = await this.prisma.transaction.create({ data })
    const validatedQuery = await this.validateTransaction(transaction)
    const signedRawTx = await this.signTxByFeePayer(transaction)
    return await this.updateTransaction(transaction, signedRawTx, validatedQuery)
  } catch (e) {
    const msg = `DelegatorError: ${e.name}`
    throw new HttpException(msg, HttpStatus.BAD_REQUEST)
  }
}
```